src/hooks/useResourceInstance.ts

```
@@ -0,0 +1,349 @@
+          }
+        });
+
+        // Initial value already has the main resource. So, if 'main' is true, then don't add any value to the Array
+
+        let clusterPorts;
+        if (data?.detailedNetworkTopology) {
+          clusterPorts = Object.values(data.detailedNetworkTopology).reduce(
+            (accumulator: any, topologyDetails: any) => {
+              if (topologyDetails.main) return accumulator;
+              return [
+                ...accumulator,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)